packages/changelog/src/index.ts

```diff
@@ -171,67 +171,127 @@ const applyChanges = ({
+            tag?: string | undefined;
           }[];
         } = {};
 
+        // **IMPORTANT**: prefill the `groupedChangesByTag` with tags from config
+        // in the same order they were defined
+        Object.keys(config.changeTags ?? {}).forEach(
+          (k) => (groupedChangesByTag[k] = [])
+        );
+
         assembledChanges.releases[change.changes.name].changes
-          .filter((c) => c.meta?.dependencies)
-          .forEach((r) => {
-            r.meta!.dependencies.forEach((key) => {
-              if (!groupedDpesChanges[key]) groupedDpesChanges[key] = [];
-              groupedDpesChanges[key].push(r);
-            });
+          .filter((c) => !c.meta?.dependencies)
+          .forEach((c) => {
+            // fallback to `defaultChangeTag` if it is set, otherwise mark this change as untagged
+            const key = c.tag ?? config.defaultChangeTag;
+            if (key) {
+              if (!groupedChangesByTag[key]) groupedChangesByTag[key] = [];
+              groupedChangesByTag[key].push(c);
+            } else {
+              untaggedChanges.push(c);
+            }
           });
 
-        if (Object.keys(groupedDpesChanges).length !== 0) {
-          for (const pkg of Object.keys(groupedDpesChanges)) {
-            addition += `\n- Bumped due to a bump in \`${pkg}\``;
+        // render untagged changes freely at the top
+        for (const release of untaggedChanges) {
+          renderRelease(release);
+        }
+
+        // render tagged changes as:
+        //
+        // ### <Long tag1 name>
+        //
+        // - `1534ae12`(#12) Change1 summary
+        // - `1534ae12`(#12) Change2 summary
+        //
+        // ### <Long tag2 name>
+        //
+        // - `1534ae12`(#12) Change1 summary
+        // - `1534ae12`(#12) Change2 summary
+        //
+        // ...etc
+        for (const [tag, change] of Object.entries(groupedChangesByTag)) {
+          if (change.length !== 0) {
+            // if the specified tag is not defined in config,
+            // fallback to the short tag specified in the change file
+            const longTag = config.changeTags?.[tag] ?? tag;
+            addition += `\n### ${longTag}\n`;
+            for (const release of change) {
+              renderRelease(release);
+            }
+          }
+        }
+
+        // group dependencies bumps and render them as:
+        //
+        // ### Dependencies bumps
+        //
+        // - Bumped due to a bump in `package-a`
+        // - Bumped due to a bump in `package-b`
+        const dependencies = Object.keys(
+          assembledChanges.releases[change.changes.name].changes
+            .filter((c) => c.meta?.dependencies)
+            .reduce((acc, c) => {
+              c.meta!.dependencies.forEach((dep) => (acc[dep] = 1));
+              return acc;
+            }, {} as { [k: string]: any })
+        );
+        if (dependencies.length !== 0) {
+          addition += `\n### Dependencies bumps\n`;
```

Comment:
```suggestion
          addition += `\n### Updated Dependencies\n`;
```

I feel that "bumps" is a bit awkward here. What do you think about this instead? I believe we can now grab the version that this `dep` was bumped too. I think that would be incredibly helpful to do something like `- dep@version` and possibly link to it somehows.

Reply:
What about dependency upates that are for external packages that are not in the monorepo?

Reply:
Huh, that is an interesting idea. Originally I was only thinking monorepo deps since that is what we did before... but idk, maybe?

I believe that by the time we get to changelog, everything should be bumped if I recall correctly. When I did the last round of updates, I was trying to keep this use case in mind.

Reply:
> Huh, that is an interesting idea. Originally I was only thinking monorepo deps since that is what we did before... but idk, maybe?

I think we should if we switch the wording to `Updated dependencies` cause change log readers will most likely expect it to be the updated dependencies of the whole project not just the ones in the monorepo.

> I believe that by the time we get to changelog, everything should be bumped if I recall correctly. When I did the last round of updates, I was trying to keep this use case in mind.

I don't think so but maybe I missed something? cause I only see previous version here https://github.com/jbolda/covector/blob/main/packages/apply/src/parents.ts#L89

Reply:
Hmm, let's punt on my thought about the version. What about `Internal Dependencies` instead?